functions/dataManager/collectionManager.ts

```diff
@@ -0,0 +1,119 @@
+  let params = {
+    TableName: Config.table,
+    KeyConditionExpression: `#partitionKey = :partitionKey AND begins_with(sortKey, :sortKey)`,
+    ExpressionAttributeNames: {
+        "#partitionKey": "partitionKey",
+    },
+    ExpressionAttributeValues: {
+        ":partitionKey": collection.partitionKey,
+        ":sortKey": `[CollectionItem]#[${collection.collectionType}]#[GameItem]`
+    }
+  };
+  
+  try {
+    paginatedData = [];
+    await getPaginatedData(params);     
+    let gameList = [] as any
+    if (paginatedData.length > 0) {
+      for (let game of paginatedData) {
+        let returnedGame = await Common.deserializeDynamoCollection(game);
+        gameList.push(returnedGame);
+      };     
+    }
+    return gameList; 
+  } catch (err: any) {
+    throw err;
+  }
+}
+
+export async function getAllCollectionItems(collectionType: string) {
+  let params = {
+    TableName: Config.table,
+    IndexName: "itemTypeIndex",
+    KeyConditionExpression: "#itemType = :itemType",
+    ExpressionAttributeNames: {
+      "#itemType": "itemType"
+    },
+    ExpressionAttributeValues: {
+      ":itemType": `[CollectionItem]#[${collectionType}]#[GameItem]`
+    }
+  };
+  
+  try {
+    paginatedData = [];
+    await getPaginatedData(params);
+    if (paginatedData.length > 0) {
+      for (let game of paginatedData) {
+        let returnedGame = await Common.monitorRunningPrice(game);
+        await Common.sendRunningPriceNotification(returnedGame);
+      };     
+    }
+  } catch (err: any) {    
+    throw err;
+  }
+}
+
+export async function addGameToCollection(game: Game, collection: Collection) {
+  try {
+    let response = await createGame(game);
+    await createGamePriceData(game);
+    let updatedCollection = await getCollection(collection);
+    return updatedCollection;
+  } catch (err: any) {
+    throw err;
+  }
+}
+
+export async function modifyGameInCollection(game: Game, collection: Collection) {
+  try {
+    let response = await modifyGame(game);
+    await modifyGamePriceData(game);
+    let updatedCollection = await getCollection(collection);
+    return updatedCollection;
+  } catch (err: any) {
+    throw err;
+  }
+}
+export async function removeGameFromCollection(game: Game, collection: Collection) {
+  try {
+    let response = await deleteGame(game);
+    await deleteGamePriceData(game);
+    let updatedCollection = await getCollection(collection);
+    return updatedCollection;
+  } catch (err: any) {
+    throw err;
+  }
+}
+
+async function getPaginatedData(params: Interfaces.ICollecionParams) : Promise<Function | undefined> {
+  try {
+    let response = await Config.docClient.query(params).promise();
+    if(response['Items'].length > 0) {
+      paginatedData = [...paginatedData, ...response['Items']];
+    }
+    if (response.LastEvaluatedKey) {
+      params.ExclusiveStartKey = response.LastEvaluatedKey;
+      return await getPaginatedData(params);
```

Comment:
I did test it before and it worked, so I thought it was good.  I had 4 games in the table and set the "Limit" attribute to 1, and it would get 1 game, add it to the array, and then get the next one correctly.  Not sure if I'm missing something there, but I'll add some unit tests for this as well.